src/utils/FillUtils.ts

```
@@ -98,25 +106,40 @@ export function getLastMatchingFillBeforeBlock(
   ) as FillWithBlock;
 }
 
-export function getFillDataForSlowFillFromPreviousRootBundle(
+export async function getFillDataForSlowFillFromPreviousRootBundle(
   latestMainnetBlock: number,
   fill: FillWithBlock,
   allValidFills: FillWithBlock[],
   hubPoolClient: HubPoolClient,
+  spokePoolClientsByChain: SpokePoolClientsByChain,
   chainIdListForBundleEvaluationBlockNumbers: number[]
 ) {
+  // Can use spokeClient.queryFillsForDeposit(_fill, spokePoolClient.eventSearchConfig.fromBlock)
+  // if allValidFills doesn't contain the deposit's first fill to efficiently find the first fill for a deposit.
+  // Note that allValidFills should only include fills later than than eventSearchConfig.fromBlock.
+
   // Find the first fill chronologically for matched deposit for the input fill.
-  const firstFillForSameDeposit = allValidFills.find(
-    (_fill: FillWithBlock) => filledSameDeposit(_fill, fill) && isFirstFillForDeposit(_fill as Fill)
+  const allMatchingFills = sortEventsAscending(
+    allValidFills.filter((_fill: FillWithBlock) => filledSameDeposit(_fill, fill))
   );
+  let firstFillForSameDeposit = allMatchingFills.find((_fill) => isFirstFillForDeposit(_fill));
 
-  // If there is no first fill for the same deposit, then throw an error.
-  if (firstFillForSameDeposit === undefined) {
```

The main benefit of this PR is that this error can now be resolved by trying to load historical fills instead of default throwing if a fill can't be found